sim/lib/core/include/core/ics/component.h

```diff
@@ -0,0 +1,20 @@
+#ifndef BENTOBOX_COMPONENT_H
+#define BENTOBOX_COMPONENT_H
+
+#include <concepts>
+
+namespace ics {
+    struct BaseComponent {
+        bool isActive = true;
+        bool operator==(const BaseComponent&) const = default;
+    };
+
+    template<class C>
+    concept Component = std::derived_from<C, BaseComponent>;
+
+    struct UnknownComponent : BaseComponent { };
+
+    static_assert(Component<UnknownComponent>);
```

Comment:
My concern is that this clutters the namespace with `UnknownComponent`, which is not used anywhere else other than this test usecase, and should thus be buried somewhere in tests. It should as a red line in tests too right?

Reply:
I think the Unknown Component is a remnant of my old implementation as well. I will remove it.

Regarding tests, if I put this in a test, it will show a red line below, however, no other test will run because the tests won't compile.

Reply:
Unknown component removed in 382e961